lib/metasploit/framework/login_scanner/ssh.rb

```diff
@@ -152,6 +152,14 @@ def gather_proof
                   if proof =~ /Version:(?<os_version>.+).+HW: (?<hardware>)/mi
                     proof = "Model: #{hardware}, OS: #{os_version}"
                   end
+                # Windows
+                elsif proof =~ /is not recognized as an internal or external command/
+                  proof = ssh_socket.exec!("systeminfo\n").to_s
+                  /OS Name:\s+(?<os_name>.+)$/ =~ proof
```

Comment:
These Perl-style regexes are a little weird on their own lines, but the whole file's strange. Leaving it alone this time...